### PR TITLE
Improve achitecture documentation

### PR DESCRIPTION
**Pull request by achimweigel**
**What this PR does / why we need it**:

Improve achitecture documentation. 

**Which issue(s) this PR fixes**:
Fixes #

**Special notes for your reviewer**:

**Release note**:

```feature user
NONE
```

### CODE REVIEW

File: docs/architecture/architecture-and-security.md [hunk 1]
```diff
@@ -73,29 +78,195 @@ The following image gives a more detailed overview how Landscaper instances are
...
-  Resource-Shoot-Cluster
-- a new namespace on one of the Target-Shoot-Clusters for a new Landscaper instance
-- a Landscaper deployment on the Target-Shoot-Cluster in the newly created namespace
+The LaaS furthermore creates two [Targets](https://github.com/gardener/landscaper/blob/master/docs/usage/Targets.md)
+as input for the Installation containing the credentials
 
-The LaaS furthermore creates two [Targets](https://github.com/gardener/landscaper/blob/master/docs/usage/Targets.md) 
-as input for the Installation containing the credentials 
+- for a Target-Shoot-Cluster. These credentials are fetched from one of the ServiceTargetConfig custom resources and 
+  are used to install all required artefacts for the Landscaper Instance on a Target-Shoot-Cluster.
```
Comment (enrico-kaack-comp) [inline docs/architecture/architecture-and-security.md:91]: and the new sidecar service controllers

File: docs/architecture/architecture-and-security.md [hunk 2]
```diff
@@ -73,29 +78,195 @@ The following image gives a more detailed overview how Landscaper instances are
...
+which are deploying:
+
+- **[shoot](../../.landscaper/landscaper-instance/blueprint/installation/shoot-cluster-subinst.yaml)**: 
+  A new shoot custom resource in the Garden-Resource-Cluster-Project for which the Gardener creates a new 
+  Resource-Shoot-Cluster
+- **[landscaper](../../.landscaper/landscaper-instance/blueprint/installation/landscaper-deployment-subinst.yaml)**: 
+  A new Landscaper and its deployers in new namespace on one of the Target-Shoot-Clusters. This Landscaper is watching 
+  and processing the Landscaper resources on the Resource-Shoot-Cluster.
+- **[ls-service-target-shoot-sidecar-server](../../.landscaper/landscaper-instance/blueprint/installation/sidecar-subinst.yaml)**: 
+  Two controllers handling the access rights of the customer/users to the Resource-Shoot-Cluster as well as the
```
Comment (enrico-kaack-comp) [inline docs/architecture/architecture-and-security.md:112]: Two controller on the Target-Shoot-Cluster

File: docs/usage/LandscaperDeployments.md [hunk 3]
```diff
@@ -26,6 +26,9 @@ spec:
       - helm
       - manifest
       - container
+  oidcConfig: # optional
+    clientID: <some client ID>
+    issuerURL: <OIDC token issuer URL>
```
Comment (enrico-kaack-comp) [inline docs/usage/LandscaperDeployments.md:31]: userClaim and groupClaim is missing I think

**Review by enrico-kaack-comp — CHANGES_REQUESTED** [review]

File: docs/architecture/architecture-and-security.md (hunk 1 above)
Comment (achimweigel) [reply docs/architecture/architecture-and-security.md:91]: the new sidecar is part of a landscaper instance (one of five sub installations) :-)

File: docs/architecture/architecture-and-security.md (hunk 2 above)
Comment (achimweigel) [reply docs/architecture/architecture-and-security.md:112]: I have changed it

File: docs/usage/LandscaperDeployments.md (hunk 3 above)
Comment (achimweigel) [reply docs/usage/LandscaperDeployments.md:31]: very good point

**Review by enrico-kaack-comp — APPROVED** [review]

**Pull request merged**